core/chains/evm/gas/rollups/l1_oracle.go

```diff
@@ -47,21 +49,21 @@ func IsRollupWithL1Support(chainType chaintype.ChainType) bool {
 	return slices.Contains(supportedChainTypes, chainType)
 }
 
-func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType) (L1Oracle, error) {
+func NewL1GasOracle(lggr logger.Logger, ethClient l1OracleClient, chainType chaintype.ChainType, daOracle evmconfig.DAOracle) (L1Oracle, error) {
 	if !IsRollupWithL1Support(chainType) {
 		return nil, nil
 	}
 	var l1Oracle L1Oracle
 	var err error
-	switch chainType {
-	case chaintype.ChainOptimismBedrock, chaintype.ChainKroma, chaintype.ChainScroll, chaintype.ChainMantle, chaintype.ChainZircuit:
-		l1Oracle, err = NewOpStackL1GasOracle(lggr, ethClient, chainType)
-	case chaintype.ChainArbitrum:
+	switch daOracle.OracleType() {
+	case toml.OPOracle:
+		l1Oracle, err = NewOpStackL1GasOracle(lggr, ethClient, chainType, daOracle)
+	case toml.ArbitrumOracle:
 		l1Oracle, err = NewArbitrumL1GasOracle(lggr, ethClient)
-	case chaintype.ChainZkSync:
+	case toml.ZKSyncOracle:
 		l1Oracle = NewZkSyncL1GasOracle(lggr, ethClient)
 	default:
-		return nil, fmt.Errorf("received unsupported chaintype %s", chainType)
+		return nil, fmt.Errorf("unsupported DA oracle type %s", daOracle.OracleType())
```

Comment:
This change was made per @matYang 's suggestion here: https://github.com/smartcontractkit/chainlink/pull/14599#discussion_r1779348577

The one "gotcha" with this is that now we need to make sure the `DAOracle` config is correctly set and non-nil for each of these chains.

Reply:
I see how this simplifies this switch but I'm worried about introducing another config for this when `ChainType` is already used to group behavior for different chains. Don't think this saves us from making changes either. If we introduce a new ChainType for a new chain, we have to make code changes like this for it anyways.

Reply:
the idea is that chaintype can deviate from oracle type, e.g if we have the heuristic oracle, many chain types may be able to share the same oracle type